src/pymovements/dataset/dataset_definition.py

```diff
@@ -181,9 +263,83 @@ class DatasetDefinition:
     acceleration_columns: list[str] | None = None
     distance_column: str | None = None
 
-    _has_resources: _HasResourcesIndexer = field(
-        default_factory=_HasResourcesIndexer, init=False, repr=False, compare=False, hash=False,
-    )
+    def __init__(
```

Comment:
Unfortunately we now need to define an explicit `__init__()`, as we want to

1. make sure that `DatasetDefinition.resources` is always of type `Resources`
2. be permissive in the input parameter type (i.e. including all `ResourceLike` datastructures)

This requires different typehints for input parameters and fields, which is not possible to define using `dataclasses.field` alone.